Settings object

MatthewFoster02 opened the pull request:
Settings object implemented closes #18

S010MON commented on src/main/java/app/controller/Settings.java:
```diff
@@ -0,0 +1,130 @@
+    private double sprintSpeed;
+    private double timeStep;
+    private double scaling;
+    private Rectangle2D[] walls;
```
Consider making these `ArrayLists` where you can keep adding them as you parse the file, if you make the getter return a copy, rather than the reference you can keep it as immutable.

S010MON commented on src/main/java/app/controller/Settings.java:
```diff
@@ -0,0 +1,130 @@
+    private Rectangle2D.Double spawnAreaGuards;
+
+    public Settings(int gamemode, int height, int width, int noOfGuards, int noOfIntruders, double walkSpeed, double sprintSpeed, double timeStep, double scaling, Rectangle2D[] walls, Rectangle2D[] shade, Rectangle2D[] towers, Rectangle2D[] portals, Rectangle2D[] textures, Rectangle2D.Double targetArea, Rectangle2D.Double spawnAreaIntruders, Rectangle2D.Double spawnAreaGuards)
+    {
```
Perhaps create a default constructor and update each element when the line is parsed, otherwise you'll have to hold all of these elements in memory until the object is created.

It also has the secondary benefit of you not needing to remember the order of all the parameters

S010MON reviewed (changes requested):
Look at using ArrayLists rather than arrays (I know i probably didn't make it very clear that that was also an option in the UML ... I was lazy and didn't want to fill everything with `ArrayList<Rectangle2d>`

Look at reducing the number of parameters you have in the constructor.  If you want to keep the class immutable, you could have a `lock()` method that allows you to modify it until you call lock which latches a boolean variable to false and stops you from adding any more data

```
boolean unlocked = true;

// This keeps working until unlocked is false
public void addWall(Rectangle2d rect)
{
    if(unlocked)
        walls.add(rect);
}

// Latching method that doesn't allow unlocking
public void lock()
{
    unlocked = false;
}
```

MatthewFoster02 commented:
I didn't see an option to re-pull request so clicked on resolve conversation. Removed the constructor and added the lock feature, maybe I went too far the opposite way with the setters but let me know if there is more to change. Thanks

S010MON commented:
So I'd agree little bit overboard with the Set methods, you could initialise empty arraylists in the constructors and get rid of them, would solve a lot of boilerplate code
